homeassistant/components/google_drive/sensor.py

```diff
@@ -0,0 +1,123 @@
+    GoogleDriveSensorEntityDescription(
+        key="storage_total",
+        translation_key="storage_total",
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.limit,
+    ),
+    GoogleDriveSensorEntityDescription(
+        key="storage_used",
+        translation_key="storage_used",
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.usage,
+    ),
+    GoogleDriveSensorEntityDescription(
+        key="storage_used_in_drive",
+        translation_key="storage_used_in_drive",
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.usage_in_drive,
+        entity_registry_enabled_default=False,
+    ),
+    GoogleDriveSensorEntityDescription(
+        key="storage_used_in_drive_trash",
+        translation_key="storage_used_in_drive_trash",
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIBIBYTES,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.usage_in_trash,
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: GoogleDriveConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up GoogleDrive sensor based on a config entry."""
+    coordinator = entry.runtime_data
+    async_add_entities(
+        GoogleDriveSensorEntity(coordinator, description)
+        for description in SENSORS
+        if description.exists_fn(coordinator.data)
+    )
+
+
+class GoogleDriveSensorEntity(GoogleDriveEntity, SensorEntity):
+    """Defines a Google Drive sensor entity."""
+
+    entity_description: GoogleDriveSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: GoogleDriveDataUpdateCoordinator,
+        description: GoogleDriveSensorEntityDescription,
+    ) -> None:
+        """Initialize a Google Drive sensor entity."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self._attr_unique_id = (
+            f"{_slugify(coordinator.config_entry.unique_id)}_{description.key}"
```

Comment:
I think I'd be okay with just the unique id in here, or I am missing something

Reply:
We had a discussion about that earlier: https://github.com/home-assistant/core/pull/156167#discussion_r2514154966